src/Components/Pages/Plot/JSRootGraph.js

```diff
@@ -1,17 +1,58 @@
-import React, { createRef } from "react";
+import React, { createRef } from 'react';
+import { StoppingPowerUnits } from '../../../Backend/Utils';
 
-let JSROOT
-//#region Helper functions
+function power10Max(values) {
+    return Math.pow(10, Math.ceil(Math.log10(Math.max(...values))))
+}
+
+function power10Min(values) {
+    return Math.pow(10, Math.floor(Math.log10(Math.min(...values))))
+}
+
+function checkBoundaries(dataSeries) {
+    if(dataSeries.length < 1) return {
+        minEnergy: 0,
+        maxEnergy: 10,
+        minValue: 0,
+        maxValue: 10,
+    }
+    // We're not handling such big energies so it's safe to assume everything will be smaller than 1e+10
+    const boundaries = {
+        minEnergy: 1e+10,
+        maxEnergy: 0,
+        minValue: 1e+10,
+        maxValue: 0,
```

Comment:
Why `minEnergy` is larger that `maxEnergy` ?

Reply:
Those are initial placeholder vales which i want to override with anything else that is actual data. We can safely assume that every value of energy that is taken from dedx will be greater than 0 and smaller than 1e+10 so no matter what series we will analyze the values will be replaces with actual output.

Reply:
Simmilarly when looking for a minimal value in an array we would use MAX_INT as a starting point beacause everything would be smaller

Reply:
I see your point, I've asked as the code was not self-explaining on that point.